### PR TITLE
api: add user delete endpoint to mark for deletion

### PR DESCRIPTION
**Pull request by kleesc**
TODO: frontend

### CODE REVIEW

File: backend/tasks/user.py [hunk 1]
```diff
@@ -7,58 +7,90 @@
...
+SECSCAN_V4_NOTIFICATION_QUEUE_NAME = "secscanv4"
+
+
+image_replication_queue = WorkQueue(REPLICATION_QUEUE_NAME, tf, has_namespace=False)
+dockerfile_build_queue = WorkQueue(DOCKERFILE_BUILD_QUEUE_NAME, tf, has_namespace=True)
+notification_queue = WorkQueue(NOTIFICATION_QUEUE_NAME, tf, has_namespace=True)
+secscan_notification_queue = WorkQueue(SECSCAN_V4_NOTIFICATION_QUEUE_NAME, tf, has_namespace=False)
+export_action_logs_queue = WorkQueue(EXPORT_ACTION_LOGS_QUEUE_NAME, tf, has_namespace=True)
+repository_gc_queue = WorkQueue(REPOSITORY_GC_QUEUE_NAME, tf, has_namespace=True)
+namespace_gc_queue = WorkQueue(NAMESPACE_GC_QUEUE_NAME, tf, has_namespace=False)
```
Comment (bcaton85) [inline backend/tasks/user.py:39]: Some of these references aren't being used, is there a reason they're included?

File: backend/tasks/user.py [hunk 2]
```diff
@@ -93,10 +125,23 @@ def put(self, username):
...
+        )
+
+    @login_required
+    def delete(self, username):
+        found_user = user.get_namespace_user(username)
+        if found_user is None:
+            return make_response(json.dumps({"message": f"Could not find user {username}"}), 404)
 
-        return make_response(json.dumps({"message": "User updated successfully", "user": username, "enabled": enable}), 200)
+        user.mark_namespace_for_deletion(found_user, all_queues, namespace_gc_queue, force=True)
```
Comment (bcaton85) [inline backend/tasks/user.py:146]: Not seeing a definition for `all_queues`, is that defined somewhere else?

**Review by bcaton85 — CHANGES_REQUESTED** [review]

File: backend/tasks/user.py (hunk 1 above)
Comment (kleesc) [reply backend/tasks/user.py:39]: Related to the other comment.

**Comment by Sunandadadi** [comment]
Can we separate the formatting into a different PR? We can merge the formatting PR without much debate. That will make this PR cleaner as well and have only the changes responsible to deleting the account.

File: backend/tasks/user.py [hunk 3]
```diff
@@ -7,58 +7,100 @@
...
+export_action_logs_queue = WorkQueue(EXPORT_ACTION_LOGS_QUEUE_NAME, tf, has_namespace=True)
+repository_gc_queue = WorkQueue(REPOSITORY_GC_QUEUE_NAME, tf, has_namespace=True)
+namespace_gc_queue = WorkQueue(NAMESPACE_GC_QUEUE_NAME, tf, has_namespace=False)
+
+
+all_queues = [
+    image_replication_queue,
+    dockerfile_build_queue,
+    notification_queue,
+    chunk_cleanup_queue,
```
Comment (bcaton85) [inline backend/tasks/user.py:46]: I think this work queue definition is missing too

Comment (kleesc) [reply backend/tasks/user.py:47]: Good catch. Fixed.

**Review by bcaton85 — APPROVED** [review]
LGTM

**Pull request merged**